app/Http/Controllers/Auth/AuthController.php

```diff
@@ -11,115 +13,155 @@
+    // to login and redirect the user back to the login form. Of course, when this
+    // user surpasses their maximum number of attempts they will get locked out.
+    if ($throttles) {
+      $this->incrementLoginAttempts ( $request );
+    }
+    
+    return redirect ( $this->loginPath () )->withInput ( $request->only ( $this->loginUsername (), 'remember' ) )->withErrors ( [ 
+        $this->loginUsername () => $this->getFailedLoginMessage () 
+    ] );
+  }
+  
+  /**
+   * Handle a registration request for the application.
+   *
+   * @param \Illuminate\Http\Request $request          
+   * @return \Illuminate\Http\Response
+   */
+  public function postRegister(Request $request) {
+    $validator = $this->validator ( $request->all () );
+    
+    if ($validator->fails ()) {
+      return redirect ( $this->registerPath )->withErrors ( $validator );
+    }
+    
+    Auth::login ( $this->create ( $request->all () ) );
+    return redirect ( $this->redirectPath () );
+  }
+  
+  /**
+   * Get a validator for an incoming registration request.
+   *
+   * @param array $data          
+   * @return \Illuminate\Contracts\Validation\Validator
+   */
+  protected function validator(array $data) {
+    return Validator::make ( $data, [ 
+        'name' => 'required|max:255|unique:users,username|min:3',
+        'email' => 'required|email|max:255|unique:users',
+        'password' => 'required|confirmed|min:8' 
+    ] );
+  }
+  public function getSocialPassword(Request $request) {
+    if (! $request->session ()->has ( 'socialUser' )) {
+      return redirect ()->intended ( '/login' );
+    }
+    return view ( 'auth.set_social_password' );
+  }
+  public function postSocialPassword(Request $request) {
+    if (Auth::check ()) {
+      return redirect ()->intended ( '/' );
+    }
+    $validation = Validator::make ( $request->all (), [ 
+        'password' => 'required|confirmed|min:8',
+        'name' => 'required|max:255|unique:users,username|min:3' 
+    ] );
+    
+    if ($validation->fails ()) {
+      return redirect ()->back ()->withInput ()->withErrors ( $validation->errors () );
+    } else {
+      $user_array = array (
+          'email' => $request->session ()->get ( 'socialUser' )->getEmail (),
+          'name' => $request->all () ['name'],
+          'password' => $request->all () ['password'] 
+      );
+      
+      Auth::login ( $this->create ( $user_array ) );
+      $request->session ()->forget ( 'socialUser' );
+      return redirect ( $this->redirectPath () );
+    }
+  }
+  /**
+   * Create a new user instance after a valid registration.
+   *
+   * @param array $data          
+   * @return User
+   */
+  protected function create(array $data) {
+    return User::create ( [ 
+        'email' => $data ['email'],
+        'username' => $data ['name'],
+        'password' => bcrypt ( $data ['password'] ),
+        'status' => TRUE,
+        'profile_state' => FALSE 
+    ] );
+  }
+}
+
+
```

Comment:
getSocialPassword, postSocialPassword and Validator functions looks like they are competing for space. Please give them want them want..Space!!!!